src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserStateObject.cs

```diff
@@ -3031,12 +3030,17 @@ internal Task WriteByteArray(byte[] b, int len, int offsetBuffer, bool canAccumu
                         int remainder = _outBuff.Length - _outBytesUsed;
 
                         // write the remainder
-                        Buffer.BlockCopy(b, offset, _outBuff, _outBytesUsed, remainder);
+                        Span<byte> copyTo = _outBuff.AsSpan(_outBytesUsed, remainder);
+                        ReadOnlySpan<byte> copyFrom = b.Slice(0, remainder);
+
+                        Debug.Assert(copyTo.Length == copyFrom.Length, $"copyTo.Length:{copyTo.Length} and copyFrom.Length{copyFrom.Length:D} should be the same");
```

Comment:
In what cases is this Debug Assert useful?

Reply:
They were useful in debugging to make sure that i was actually copying the amount of data i thought i should be. If you don't want to keep them they can be removed.

Reply:
Leave them around. If they were useful to you while debugging, then they could be useful in future as well.